vizro-core/src/vizro/actions/_filter_action.py

```diff
@@ -33,7 +33,7 @@ def _filter(
     """
     return _get_modified_page_figures(
         targets=targets,
-        ctds_filter=ctx.args_grouping["filters"],
-        ctds_filter_interaction=ctx.args_grouping["filter_interaction"],
-        ctds_parameters=ctx.args_grouping["parameters"],
+        ctds_filter=ctx.args_grouping["external"]["filters"],
```

Comment:
"external" dictionary key is used for handling both pre-defined and custom actions. 
"internal" dictionary key is used for handling internal inputs and outputs like:
- `{"action_finished": Output("action_finished", "data", allow_duplicate=True)},` or
- ` {"trigger": Input({"type": "action_trigger", "action_name": self.id}, "data")}`

This structure helps us to avoid "juggling" (e.g. removing action_finished as `outputs.remove("action_finished")`) in the callback_wrapper. 

This change doesn't affect users (so it can be changed any time) and only helps us not to worry about `"hardcoded"` inputs and outputs that exist for every action (pre-defined or custom).

Reply:
Thanks for explaining! That makes sense 👍